subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/verification/verifier/DependencyVerifier.java

```diff
@@ -106,10 +106,17 @@ private void doVerifyArtifact(ModuleComponentArtifactIdentifier foundArtifact, C
             for (ArtifactVerificationMetadata verification : verifications) {
                 String verifiedArtifact = verification.getArtifactName();
                 if (verifiedArtifact.equals(foundArtifactFileName)) {
-                    if (signature == null && config.isVerifySignatures()) {
-                        builder.failWith(new MissingSignature(file));
-                    }
-                    if (signature != null) {
+                    if (signature == null) {
```

Comment:
This whole method could get some love and be refactored in smaller methods. 

But I didn't want to do a refactoring with the fix to not introduce some additional bug.